tests/manifold-map.cc

```diff
@@ -356,39 +407,41 @@ BOOST_AUTO_TEST_CASE_TEMPLATE (manifold_map_test_3, T, functionTypes_t)
 
   typename Instance_H_On_MultipleReal3::argument_t input = Eigen::VectorXd::Zero(6 * static_cast<long>(posNumber));
   typename Instance_H_On_MultipleReal3::result_t result = Eigen::VectorXd::Zero(3);
-  typename Instance_H_On_MultipleReal3::gradient_t gradient = Eigen::VectorXd::Zero(6 * static_cast<long>(posNumber));
-  typename Instance_H_On_MultipleReal3::jacobian_t jacobian = Eigen::MatrixXd::Zero(3, 6 * static_cast<long>(posNumber));
+  typename Instance_H_On_MultipleReal3::gradient_t gradient(6 * static_cast<int> (posNumber));
+  gradient.setZero();
+  typename Instance_H_On_MultipleReal3::jacobian_t jacobian(3, 6 * static_cast<int> (posNumber));
+  jacobian.setZero();
 
   Instance_H_On_MultipleReal3 instWrap(descWrapPtr, problemManifold, problemManifold, reals, restrictions);
 
- for (int i = 0; i < input.size(); ++i)
+  for (int i = 0; i < input.size(); ++i)
     {
       input(i) = (i % 6) > 2;
     }
 
- instWrap(result, input);
+  instWrap(result, input);
 
- (*output) << instWrap;
+  (*output) << instWrap;
 
- std::cout << "result: " << result << std::endl;
+  std::cout << "result: " << result << std::endl;
 
- BOOST_CHECK (result(0) == 15);
- BOOST_CHECK (result(1) == 15);
- BOOST_CHECK (result(2) == 15);
+  BOOST_CHECK (result(0) == 15);
+  BOOST_CHECK (result(1) == 15);
+  BOOST_CHECK (result(2) == 15);
 
- instWrap.jacobian(jacobian, input);
+  instWrap.jacobian(jacobian, input);
 
- Eigen::MatrixXd jac = jacobian;
+  Eigen::MatrixXd jac = jacobian;
 
- (*output) << "\n";
+  (*output) << "\n";
 
- for (int i = 0; i < jac.rows(); ++i)
-   {
-     std::cout << jac.row(i) << std::endl;
-     (*output) << jac.row(i) << std::endl;
-   }
+  for (int i = 0; i < jac.rows(); ++i)
+    {
+      std::cout << jac.row(i) << std::endl;
```

Comment:
Why print to `std::cout` like this rather than doing the usual `std::cout << output->str () << std::endl;` at the end?